### DISCUSSION TITLE
Identical Power Readings for Three Phases in Solarman Integration with Home Assistant

### DISCUSSION THREAD

**Discussion by zampix1** [StephanJoubert/home_assistant_solarman#678] [Q&A]
I'm encountering an issue with the Solarman integration in HA where the power readings for the three phases (R, S, and T) are identical, despite being configured to read from different Modbus registers. According to the inverter’s Modbus documentation, each phase should have a unique power reading, but the values returned are consistently the same across all phases.

Technical Details:

    Inverter Model: KSTAR E10KT Clone EEI E-CASA 10 HU  (Modbus RS485 protocol)
    Modbus Registers Configured for Each Phase:
        R-phase Grid Power: Register 3100
        S-phase Grid Power: Register 3104
        T-phase Grid Power: Register 3108
    Environment: HA with YAML-based Modbus configuration kstar_hybrid.yaml using the Solarman integration.

Steps Taken to Troubleshoot:

    YAML Configuration Check: Verified that the registers are correctly assigned in the YAML configuration based on the inverter's Modbus protocol documentation.
    Staggered Readings: Collected power readings for each phase a few seconds apart. Despite this, all three phases show nearly identical values.
    Graphical Evidence: Attached graphs show identical patterns for all three phases over time, strongly suggesting that the integration may be pulling data from the same source for each phase or duplicating values.
    
    

Possible Causes Considered:

    Duplicate Register Readings: The integration might be reading from the same register across all phases, even though different registers are specified.
    Polling or Communication Bug: There could be an issue in the polling mechanism or a bug within the integration that leads to duplicated values for each phase.
    

Request for Support:
I would appreciate any guidance on how to ensure distinct readings for each phase. If there's a way to test or verify the registers being polled directly, please let me know. Additionally, if anyone has encountered similar issues or has insights on potential configuration adjustments, your assistance would be invaluable.

Attached Evidence:
Please refer to the attached graphs, which clearly show identical power values for R, S, and T phases over an extended period.

![immagine2](https://github.com/user-attachments/assets/c2e30942-b9ee-4200-baf1-f8d24b593ddf)
![immagine1](https://github.com/user-attachments/assets/4590dc9c-e07a-4c74-8f94-e1b8a0611126)
![immagine3](https://github.com/user-attachments/assets/4e709f4a-3f68-4352-9661-5c185d616f5c) 
![immagine](https://github.com/user-attachments/assets/4d23f4c6-927a-466b-a0e2-f191a0cd22f2)
![immagine](https://github.com/user-attachments/assets/48c5b484-b719-4db4-98fe-b2f4e9e95966)

Thank you in advance for any assistance provided!

**Comment by davidrapan** [comment]
Hi @zampix1, I would say that this is question for your manufacturer. If you did not make any errors and really properly configured yaml to read from 3100, 3104 and 3108. That means those registers contains incorrect values (I don't see any reason why reading on the integration side should be broken in this specific case).
